skyportal/handlers/api/instrument.py

```diff
@@ -25,8 +39,32 @@ def post(self):
                 'Invalid/missing parameters: ' f'{exc.normalized_messages()}'
             )
         instrument.telescope = telescope
-        DBSession().add(instrument)
-        self.verify_and_commit()
+
+        try:
+            instrument = (
+                Instrument.query_records_accessible_by(
+                    self.current_user,
+                )
+                .filter(
+                    Instrument.name == instrument.name,
+                    Instrument.telescope == instrument.telescope,
+                )
+                .one()
+            )
+        except NoResultFound:
+            DBSession().add(instrument)
+            self.verify_and_commit()
```

Comment:
```suggestion
        instrument = (
            Instrument.query_records_accessible_by(
                self.current_user,
            )
            .filter(
                Instrument.name == instrument.name,
                Instrument.telescope == instrument.telescope,
            )
            .first()
        )
        if instrument is not None:
            DBSession().add(instrument)
            self.verify_and_commit()
```
but why add this check at all? why wasn't it needed before, but is now?

Reply:
So for example, ZTF is already in the database, and this will simply add the fields to it.